x-pack/legacy/plugins/siem/public/hooks/api/helpers.tsx

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IndexPatternSavedObject } from '../types';
+
+/**
+ * Returns a string array of Index Pattern Titles
+ *
+ * @param indexPatterns IndexPatternSavedObject[] as provided from the useIndexPatterns() hook
+ */
+export const getIndexPatternTitles = (indexPatterns: IndexPatternSavedObject[]): string[] =>
+  indexPatterns.reduce((acc: string[], v) => [...acc, v.attributes.title], []);
```

Comment:
You want to use the regular template like you did above:

```suggestion
  indexPatterns.reduce<string[]>((acc, v) => [...acc, v.attributes.title], []);
```